common-service-core/src/main/java/org/zowe/apiml/util/MapUtils.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.util;
+
+import java.util.*;
+import java.util.function.Consumer;
+
+public class MapUtils {
+
+    public Map<String, String> flattenMap(String rootKey, Map<String, Object> collection) {
+        if (collection == null || collection.isEmpty()) {
+            return Collections.emptyMap();
+        }
+
+        Map<String, String> result = new HashMap<>();
+
+        Map<Class, Consumer<Map.Entry<String, Object>>> actionMap = new LinkedHashMap<>();
+        actionMap.put(Map.class, entry -> result.putAll(flattenMap(mergeKey(rootKey, entry.getKey()), (Map<String, Object>)entry.getValue())));
+        actionMap.put(String.class, entry -> result.put(mergeKey(rootKey, entry.getKey()), entry.getValue().toString()));
+        actionMap.put(Boolean.class, entry -> result.put(mergeKey(rootKey, entry.getKey()), entry.getValue().toString()));
+        actionMap.put(Integer.class, entry -> result.put(mergeKey(rootKey, entry.getKey()), entry.getValue().toString()));
+        actionMap.put(Double.class, entry -> result.put(mergeKey(rootKey, entry.getKey()), entry.getValue().toString()));
+        actionMap.put(Float.class, entry -> result.put(mergeKey(rootKey, entry.getKey()), entry.getValue().toString()));
+        actionMap.put(List.class, entry -> { throw new IllegalArgumentException("List parsing is not supported"); } );
+        actionMap.put(Object[].class, entry -> { throw new IllegalArgumentException("Array parsing is not supported"); } );
+        actionMap.put(Object.class, entry -> { throw new IllegalArgumentException(String.format("Cannot parse key: %s with value %s", entry.getKey(), entry.getValue().toString())); } );
+
+        for (Map.Entry<String, Object> entry : collection.entrySet()) {
+            if (entry.getValue() == null) {
+                result.put( mergeKey(rootKey, entry.getKey()), "");
+                continue;
+            }
+            executeAction(entry,actionMap);
+        }
+        return result;
+    }
+
+    private void executeAction(Map.Entry<String, Object> switchSubject, Map<Class, Consumer<Map.Entry<String, Object>>> actionMap) {
+        for (Map.Entry<Class, Consumer<Map.Entry<String, Object>>> action : actionMap.entrySet()) {
```

Comment:
There is possible to test in first step with current class. If it is not found, then make this iteration and then store new type in the class. It could be helpfull (like cache), but I am not sure how often it is run.

Map<Class, Consumer<Map.Entry<String, Object>>> actionMap = new LinkedHashMap<>();
Map<Class, Consumer<Map.Entry<String, Object>>> cache = new HashMap<>();
static {
init(actionMap);
cache.putAll(actionMap)
}

....

Consumer c = cache.get(...);
if (c != null) return c....;

for (Entry e : actionMap) {
if (instanceof...) {
cache.put(class, e.getValue);
return e.getValue....
}

Reply:
check out the latest impl